package.json

```diff
@@ -49,6 +49,7 @@
     "devtools-protocol": "0.0.781568",
     "extract-zip": "^2.0.0",
     "https-proxy-agent": "^4.0.0",
+    "socks-proxy-agent": "^5.0.0",
```

Comment:
I actually am not sure what to do here Found few modules.

1. https://www.npmjs.com/package/socks-proxy-agent Similar to current Https module
2. https://www.npmjs.com/package/proxy-agent Seems versatile can be used instead of 2 seperate modules.
3. https://www.npmjs.com/package/simple-proxy-agent Low # of downloads
4. https://www.npmjs.com/package/socks-proxy-agent-ts Even lower # of downloads

I've used 1st but 2nd seems better suited (PAC support et.al.) So directly pass logic to that module entirely.

Reply:
Given that `proxy-agent` already implements all the different protocols and resolving ENV variables it sounds like the best option to me.